### ISSUE TITLE
"change [variable] by (1)" block does not result in NaN when manipulating strings

### ISSUE THREAD

**Issue by thisandagain** [LLK/scratch-vm#1879]
### Expected Behavior
Executing the "change [variable] by (1)" block on a variable that is currently set to a string should result in `NaN`

### Actual Behavior
The variable is set to `0`

### Steps to Reproduce
![image](https://user-images.githubusercontent.com/747641/50519879-f3967800-0a8a-11e9-9314-b2d5d0f6f894.png)

### Reference
https://scratch.mit.edu/projects/275276918
https://llk.github.io/scratch-gui/develop/#275276918

Major HT to Scratcher [hahasamian](https://scratch.mit.edu/users/hahasamian/) for identifying this issue!

/cc @kchadha @ktbee

**Comment by apple502j** [comment]
Wait, compat wiki mentions this.
https://github.com/LLK/scratch-vm/wiki/Compatibility#consistent-type-coercion-when-modifying-variables
Should it be fixed or kept? @thisandagain

**Comment by thisandagain** [comment]
@apple502j Yeah. This was one of the very first things we decided to change. I'd like to discuss this.
